libethereum/ClientTest.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file ClientTest.cpp
+ * @author Dimitry Khokhlov <dimitry@ethdev.com>
+ * @date 2016
+ */
+
+#include <libethereum/EthereumHost.h>
+#include <libethereum/ClientTest.h>
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+using namespace p2p;
+
+ClientTest& dev::eth::asClientTest(Interface& _c)
+{
+	return dynamic_cast<ClientTest&>(_c);
+}
+
+ClientTest* dev::eth::asClientTest(Interface* _c)
+{
+	return &dynamic_cast<ClientTest&>(*_c);
+}
+
+DEV_SIMPLE_EXCEPTION(ChainParamsNotEthash);
+
+ClientTest::ClientTest(
+	ChainParams const& _params,
+	int _networkID,
+	p2p::Host* _host,
+	std::shared_ptr<GasPricer> _gpForAdoption,
+	std::string const& _dbPath,
+	WithExisting _forceAction,
+	TransactionQueue::Limits const& _limits
+):
+	Client(_params, _networkID, _host, _gpForAdoption, _dbPath, _forceAction, _limits)
+{
+}
+
+bool ClientTest::addBlock(string const& _rlp)
+{
+	if (auto h = m_host.lock())
+		h->noteNewBlocks();
+
+	bytes rlpBytes = fromHex(_rlp.substr(0, 2) == "0x" ? _rlp.substr(2) : _rlp, WhenError::Throw);
```

Comment:
I think the check for `"0x"` is not needed, that should be done inside `fromHex`